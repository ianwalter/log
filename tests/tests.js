const { test } = require('@ianwalter/bff')
const mockStdio = require('mock-stdio')
const signale = require('signale')
const { log, Log } = require('..')

test('logger', ({ expect }) => {
  log.update()
  mockStdio.start()
  log.info('YO!')
  expect(mockStdio.end()).toMatchSnapshot()
})

test('log level', ({ expect }) => {
  log.update()
  mockStdio.start()
  log.update({ level: 'warn' })
  log.warn('I am the warrior!')
  log.info('Moonbeam')
  expect(mockStdio.end()).toMatchSnapshot()
})

test('works with signale', ({ expect }) => {
  log.update()
  mockStdio.start()
  log.update({ types: Object.keys(signale._types), logger: signale })
  log.success('Whoop! Whoop!')
  const { stdout } = mockStdio.end()
  expect(stdout).toContain('âœ”')
  expect(stdout).toContain('Whoop! Whoop!')
})

test('creating a new log instance', ({ expect }) => {
  log.update()
  const warnLog = log.create({ level: 'warn' })
  expect(log.options.level).toBe('info')
  expect(warnLog.options.level).toBe('warn')
})

test('Log is the Log class', ({ expect }) => {
  log.update()
  expect(Log.name).toBe('Log')
  const logInstance = new Log()
  expect(logInstance.constructor.name).toBe('Log')
})

test('returns what logger returns', ({ expect }) => {
  const log = new Log({
    logger: {
      info (msg) {
        return msg
      }
    }
  })
  const msg = 'Elizabeth Warren for President'
  expect(log.info(msg)).toBe(msg)
})
