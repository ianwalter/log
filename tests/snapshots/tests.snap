// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log level 1`] = `
Object {
  "stderr": "I am the warrior!
",
  "stdout": "",
}
`;

exports[`logger 1`] = `
Object {
  "stderr": "",
  "stdout": "YO!
",
}
`;
